src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -3943,47 +3665,13 @@ private bool UsesParentsNestedAwareProperty(IExternalControl controlInfo, DName
                         controlInfo.Template.NestedAwareTableOutputs.Contains(propertyName);
             }
 
-            public static BinderCheckTypeResult PostVisitCore(ErrorContainer errorContainer, UnaryOpNode node, DType childType)
-            {
-                Contracts.AssertValue(node);
-
-                switch (node.Op)
-                {
-                    case UnaryOp.Not:
-                        var resNot = CheckTypeCore(errorContainer, node.Child, childType, DType.Boolean, /* coerced: */ DType.Number, DType.String, DType.OptionSetValue);
-                        return new BinderCheckTypeResult() { Success = true, Node = node, NodeType = DType.Boolean, Coercions = resNot.Coercions };
-                    case UnaryOp.Minus:
-                        switch (childType.Kind)
-                        {
-                            case DKind.Date:
-                                // Important to keep the type of minus-date as date, to allow D-D/d-D to be detected
-                                return new BinderCheckTypeResult() { Success = true, Node = node, NodeType = DType.Date };
-                            case DKind.Time:
-                                return new BinderCheckTypeResult() { Success = true, Node = node, NodeType = DType.Time };
-                            case DKind.DateTime:
-                                // Important to keep the type of minus-datetime as datetime, to allow d-d/D-d to be detected
-                                return new BinderCheckTypeResult() { Success = true, Node = node, NodeType = DType.DateTime };
-                            default:
-                                var resDefault = CheckTypeCore(errorContainer, node.Child, childType, DType.Number, /* coerced: */ DType.String, DType.Boolean);
-                                return new BinderCheckTypeResult() { Success = true, Node = node, NodeType = DType.Number, Coercions = resDefault.Coercions };
-                        }
-
-                    case UnaryOp.Percent:
-                        var resPercent = CheckTypeCore(errorContainer, node.Child, childType, DType.Number, /* coerced: */ DType.String, DType.Boolean, DType.Date, DType.Time, DType.DateTimeNoTimeZone, DType.DateTime);
-                        return new BinderCheckTypeResult() { Success = true, Node = node, NodeType = DType.Number, Coercions = resPercent.Coercions };
-                    default:
-                        Contracts.Assert(false);
-                        return new BinderCheckTypeResult() { Success = false, Node = node, NodeType = DType.Error };
-                }
-            }
-
             public override void PostVisit(UnaryOpNode node)
             {
                 AssertValid();
 
                 var childType = _txb.GetType(node.Child);
 
-                var res = PostVisitCore(_txb.ErrorContainer, node, childType);
+                var res = CheckUnaryOpCore(_txb.ErrorContainer, node, childType);
```

Comment:
>CheckUnaryOpCore [](http://example.com/codeflow?start=26&length=16)

BinderUtils.CheckUnaryOpCore?

Reply:
All the BinderUtils symbols are being statically imported.